File: 0027-remove-element/0027-remove-element.c
int removeElement(int* nums, int numsSize, int val) {
    int k = 0;
    int p = 0; // Pointer

    for (int i = 0; i < numsSize; i++) {
        if (nums[i] != val) {
            nums[p] = nums[i];
            p++;
            k++;
        }
    }

    return k;
}

File: 0027-remove-element/0027-remove-element.java
class Solution {
    public int removeElement(int[] nums, int val) {
        int l = 0;
        int p = 0;

        int n = nums.length;

        while(p < n) {
            if(nums[p] != val) {
                nums[l++] = nums[p++];
            } else {
                p++;
            }
        }

        return l;
    }
}